pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -475,45 +475,51 @@ func buildInboundHTTPFilter(mesh *meshconfig.MeshConfig, attrs attributes, node
 	return out
 }
 
-func addFilterConfigToRoute(in *plugin.InputParams, httpRoute *route.Route, attrs attributes) {
+func addFilterConfigToRoute(in *plugin.InputParams, httpRoute *route.Route, attrs attributes,
+	quotaSpec []*mccpb.QuotaSpec) {
 	httpRoute.TypedPerFilterConfig = addTypedServiceConfig(httpRoute.TypedPerFilterConfig, &mccpb.ServiceConfig{
 		DisableCheckCalls:  disablePolicyChecks(outbound, in.Push.Mesh, in.Node),
 		DisableReportCalls: in.Push.Mesh.GetDisableMixerHttpReports(),
 		MixerAttributes:    &mpb.Attributes{Attributes: attrs},
 		ForwardAttributes:  &mpb.Attributes{Attributes: attrs},
+		QuotaSpec:          quotaSpec,
 	})
 }
 
 func modifyOutboundRouteConfig(push *model.PushContext, in *plugin.InputParams, virtualHostname string, httpRoute *route.Route) *route.Route {
+	isPolicyCheckDisabled := disablePolicyChecks(outbound, in.Push.Mesh, in.Node)
 	// default config, to be overridden by per-weighted cluster
 	httpRoute.TypedPerFilterConfig = addTypedServiceConfig(httpRoute.TypedPerFilterConfig, &mccpb.ServiceConfig{
-		DisableCheckCalls:  disablePolicyChecks(outbound, in.Push.Mesh, in.Node),
+		DisableCheckCalls:  isPolicyCheckDisabled,
 		DisableReportCalls: in.Push.Mesh.GetDisableMixerHttpReports(),
 	})
 	switch action := httpRoute.Action.(type) {
 	case *route.Route_Route:
 		switch upstreams := action.Route.ClusterSpecifier.(type) {
 		case *route.RouteAction_Cluster:
-			_, _, hostname, _ := model.ParseSubsetKey(upstreams.Cluster)
-			var attrs attributes
-			if hostname == "" && upstreams.Cluster == util.PassthroughCluster {
-				attrs = addVirtualDestinationServiceAttributes(make(attributes), util.PassthroughCluster)
-			} else {
-				svc := in.Node.SidecarScope.ServiceForHostname(hostname, push.ServiceByHostnameAndNamespace)
-				attrs = addDestinationServiceAttributes(make(attributes), svc)
+			{
+				_, _, hostname, _ := model.ParseSubsetKey(upstreams.Cluster)
+				var attrs attributes
+				if hostname == "" && upstreams.Cluster == util.PassthroughCluster {
+					attrs = addVirtualDestinationServiceAttributes(make(attributes), util.PassthroughCluster)
+				} else {
+					svc := in.Node.SidecarScope.ServiceForHostname(hostname, push.ServiceByHostnameAndNamespace)
+					attrs = addDestinationServiceAttributes(make(attributes), svc)
+				}
+				addFilterConfigToRoute(in, httpRoute, attrs, getQuotaSpec(in, hostname, isPolicyCheckDisabled))
```

Comment:
If this applies to sidecar, does it mean there will be two rate limits for a call?

Reply:
No, it will not. The check calls decision is still determined by disablePolicyChecks(outbound, in.Push.Mesh, in.Node) which is unchanged. 
This just adds QuotaSpec to gateways too.

Reply:
For sidecar, we have added quota spec to inbound route cinfig in `buildInboundRouteConfig`

>This just adds QuotaSpec to gateways too.

Can you elaborate how do you guarantee it?

Reply:
so just quota spec doesn't imply that there will be quota calls. Quota calls are made as part of check calls. So as number of check calls are not changing, rate limit calls will not change.

What this is doing is that for gateways now, when they make a check call, they can make a rate limit call too as they have quotaspec too in it now.

Reply:
So why donot you add check before it, only call  `addFilterConfigToRoute` when node type is router